crates/router/src/core/payment_methods/cards.rs

```diff
@@ -1344,42 +1344,50 @@ pub async fn list_payment_methods(
         .await?;
     }
 
-    // Filter out applepay payment method from mca if customer has already saved it
-    response
-        .iter()
-        .position(|pm| {
-            pm.payment_method == enums::PaymentMethod::Wallet
-                && pm.payment_method_type == enums::PaymentMethodType::ApplePay
-        })
+    // Filter out wallet payment method from mca if customer has already saved it
+    customer
         .as_ref()
-        .zip(customer.as_ref())
-        .async_map(|(index, customer)| async {
-            match db
-                .find_payment_method_by_customer_id_merchant_id_list(
-                    &customer.customer_id,
-                    &merchant_account.merchant_id,
-                    None,
-                )
-                .await
-            {
-                Ok(customer_payment_methods) => {
-                    if customer_payment_methods.iter().any(|pm| {
-                        pm.payment_method == enums::PaymentMethod::Wallet
-                            && pm.payment_method_type == Some(enums::PaymentMethodType::ApplePay)
-                    }) {
-                        response.remove(*index);
-                    }
-                    Ok(())
-                }
-                Err(error) => {
-                    if error.current_context().is_db_not_found() {
+        .async_map(|customer| async {
+            let wallet_pm_exists = response
```

Comment:
what is a merchant wants to save multiple apple_pay for a customer? do we have any provision to support it ? if not let's create on github issue

Reply:
@jagan-jaya We need to solve the problem of uniquely identifying an apple-pay wallet before we can do that. Currently we don't have a unique identifier for a customer's apple-pay wallet (for cards, it's the card details themselves), mainly cuz the backend is usually not involved in the apple-pay customer payment experience.

Reply:
Paypal gets the masked cards details of the wallet it saves, which is the unique identifier. we can check it once
https://developer.paypal.com/docs/api/payment-tokens/v3/#customer_payment-tokens_get